tools/project_manager/page_detail.php

```diff
@@ -43,18 +47,61 @@
     $round_for_page_selection = null;
 }
 
-$project = new Project($projectid);
 $state = $project->state;
 $title = $project->nameofwork;
 $page_details_str = _('Page Detail');
 
 output_header("$page_details_str: $title", NO_STATSBAR);
 
+// NB We use a flexbox to position the linkbox div rather than just using
+// `float: right` like the other linkboxes in the codebase do because on
+// this page the inflexible and very wide page details table causes the
+// linkbox to display in ugly locations.
+echo "<div style='display: flex; justify-content: space-between'>";
+echo "<div>";
 echo "<h1>" . html_safe($title) . "</h1>\n";
 echo "<h2>$page_details_str</h2>\n";
+echo "</div>";
+if ($may_select_user) {
+    echo "<div>";
+    echo_filter_box($projectid, $show_image_size, $username_for_page_selection, $round_for_page_selection);
+    echo "</div>";
+}
+echo "</div>";
 
 echo "<p>" . return_to_project_page_link($projectid, ["expected_state=$state"]) . "</p>\n";
 
+
+function echo_filter_box(string $projectid, int $show_image_size, ?string $username, ?Round $round)
+{
+    $username = $username === '' ? User::current_username() : $username;
+    $roundid = $round ? $round->id : '';
+    echo "<div id='linkbox'>";
+    echo "<form action='#' method='get'>";
+    echo "<p>";
+    echo "<input type='hidden' name='project' value='$projectid'>";
+    echo "<input type='hidden' name='show_image_size' value='$show_image_size'>";
+    echo "<table>";
+    echo "<tr><td colspan=2>", _("Filter pages by:"), "</td></tr>";
+    echo "<tr>";
+    echo "<td class='bold'>", _("Username"), "</td>";
+    echo "<td><input type='text' name='select_by_user' value='" . attr_safe($username) . "' required></td>";
+    echo "</tr>";
+    echo "<tr>";
+    echo "<td class='bold'>", _("User in round"), "</td>";
+    echo "<td><input type='text' name='select_by_round' value='" . attr_safe($roundid) . "'></td>";
+    echo "</tr>";
+    echo "<tr>";
+    echo "<td colspan=2>";
+    echo "<input type='submit' value='" . attr_safe(_("Refresh")) . "'>";
```

Comment:
might be worth having a clear button for PMs that want to go back to seeing all pages.

Reply:
I don't think it's possible to do that directly within the form, because form submits explicitly put all text inputs into the query even if they're empty.

Unfortunately, `page_detail.php` distinguishes between `$select_by_user` being the empty string (ie `...&select_by_user=&...` in the URL) and null (ie absent entirely from the URL). I did have a 'clever trick' to fake this with a button-link, but I decided against it in the end.

```
function echo_button_link(string $label, string $url, bool $enabled)
{
    if ($enabled) {
        // TODO(bfoley) Move to .less
        echo "<a style='color: black; text-decoration: none' href='", attr_safe($url), "'>";
        echo "<button type='button'>$label</button>";
        echo "</a>";
    } else {
        echo "<button type='button' disabled>$label</button>";
    }
}
echo_button_link(_("Show All Pages"), '?project=$projectid&show_image_size=$show_image_size' style=', !is_null($select_by_user));
```

We can bring it back if we decide it's worth it.

Reply:
Yeah, Brian had the Show All Pages in the form initially. That didn't feel like the right place for it. We're only going to show this to PFs, Squirrels and PMs. Ordinary users won't see it, because they can't `select_by_user`, so they need the "Show just my pages" and "Show all pages instead" links  not to move from where they are, above the pages table. Hopefully, PMs are smart enough to remember to look there if they need to clear the form. That approach works well for me in the sandbox version.